src/timeman.cpp

```diff
@@ -105,27 +61,37 @@ void TimeManagement::init(Search::LimitsType& limits, Color us, int ply) {
   }
 
   startTime = limits.startTime;
-  optimumTime = maximumTime = std::max(limits.time[us], minThinkingTime);
 
-  const int maxMTG = limits.movestogo ? std::min(limits.movestogo, MoveHorizon) : MoveHorizon;
+  int mtg = limits.movestogo ? std::min(limits.movestogo, 50) : 50;
 
-  // We calculate optimum time usage for different hypothetical "moves to go" values
-  // and choose the minimum of calculated search time values. Usually the greatest
-  // hypMTG gives the minimum values.
-  for (int hypMTG = 1; hypMTG <= maxMTG; ++hypMTG)
-  {
-      // Calculate thinking time for hypothetical "moves to go"-value
-      hypMyTime =  limits.time[us]
-                 + limits.inc[us] * (hypMTG - 1)
-                 - moveOverhead * (2 + std::min(hypMTG, 40));
+  TimePoint timeLeft =  std::max(TimePoint(0),
+      limits.time[us] + limits.inc[us] * (mtg - 1) - moveOverhead * (2 + mtg));
 
-      hypMyTime = std::max(hypMyTime, TimePoint(0));
+  timeLeft = slowMover * timeLeft / 100;
 
```

Comment:
It might make sense to calculate std::log2(ply + 1) beforehand, since we use it twice in each branch.

Reply:
An array of precalculated log2 values might be faster.

Reply:
that's not really needed, this is called once per move computed. The overhead is almost 0.